rasterio/tools.py

```diff
@@ -0,0 +1,62 @@
+"""Rasterio tools module
+
+See this RFC about Rasterio tools:
+https://github.com/mapbox/rasterio/issues/1300.
+"""
+
+import json
+
+import rasterio
+from rasterio.features import dataset_features
+
+
+class JSONSequenceTool(object):
+    """Extracts data from a dataset file and saves a JSON sequence
+    """
+
+    def __init__(self, func):
+        """Initialize tool
+
+        Parameters
+        ----------
+        func : callable
+            A function or other callable object that takes a dataset and
+            yields JSON serializable objects.
```

Comment:
Love it. Higher-order programming FTW.

Returing a callable might be a bit too advanced for a first example. Would it be worth adding a simpler `hello_world` tool that would serve as a starting template?

Reply:
Yes... though because most of our interesting candidate functions require extra args or keyword args, the partial seemed the more general approach. Will have to think about this one.